macros.in

```diff
@@ -1273,12 +1273,27 @@ end}
 # -S<scm name>	Sets the used patch application style, eg '-S git' enables
 #           	usage of git repository and per-patch commits.
 # -N		Disable automatic patch application
-# -p<num>	Use -p<num> for patch application	
+# -p<num>	Use -p<num> for patch application
 %autosetup(a:b:cDn:TvNS:p:)\
 %setup %{-a} %{-b} %{-c} %{-D} %{-n} %{-T} %{!-v:-q}\
 %{-S:%global __scm %{-S*}}\
 %{expand:%__scm_setup_%{__scm} %{!-v:-q}}\
 %{!-N:%autopatch %{-v} %{-p:-p%{-p*}}}
 
+# Macro to remove or replace tilde from version.
+# -v<version> Sets the version to be used for replacement. Default is %version.
+%version_notilde(v:) %{lua:
+    local sep = rpm.expand('%1')
+    local ver = rpm.expand('%{!-v:%{version}}%{-v:%{-v*}}')
```

Comment:
Rpm expands macro arguments these days. This probably predates 4.14.0...

Reply:
so I should add more percent signs? from testing I have been doing, this macro works fine.

Reply:
Erm, never mind. This is Lua, and rpm.expand() is needed to get at the arguments in the first place.